tests/Serializer/AbstractSerializerTest.php

```diff
@@ -65,6 +65,41 @@ public function testObjectsAreNotStrings(): void
         $this->assertSame(['key' => 'value'], $result);
     }
 
+    public function traversableDataProvider(): \Generator
+    {
+        $getArray = static function (): array {
+            return [
+                'a' => 'foo',
+                'b' => 'bar',
+            ];
+        };
+
+        yield [$getArray(), new \ArrayIterator($getArray())];
+
+        // Also test with a non-rewindable non-cloneable iterator:
+        $newGenerator = static function (array $array): \Generator {
+            yield from $array;
+        };
+
+        yield [$getArray(), $newGenerator($getArray())];
+    }
+
+    /**
+     * @dataProvider traversableDataProvider
+     */
+    public function testTraversablesAreNotConsumed(array $array, \Traversable $traversable): void
+    {
+        $serializer = $this->createSerializer();
+
+        $traversed = [];
+        foreach ($traversable as $k => $v) {
```

Comment:
Not sure about what you are trying to achieve here, but it seems wrong to iterate the traversable and call for each iteration the serialization of the traversable itself: just compare the expected result with the call to `$this->invokeSerialization($serializer, iterator_to_array($traversable))`

Reply:
I'm reproducing #1028: Behat traverses the features through an iterator (taken as a function argument), and for each feature, if a scenario triggers a catchable exception (like the expected `NotFoundHttpException` in my case of testing an HTTP 404), it is captured by Sentry, which ends up serializing the stacktrace (that includes the iterator).
So here I don't care whether the serialization result is `[1, 2, 3]` or `'Object ArrayIterator'`, I just want to test that the serialization of the iterator doesn't cause the iteration to end prematurely (by default, i.e. not enabling `serializeAllObjects`). Do you see a better way to achieve that?

Reply:
>Do you see a better way to achieve that?

Nothing worth changing the test logic itself. I just asked for a few CS fixes, and then LGTM